release.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -euo pipefail
+
+VERSION="$(date +%Y.%m.%d)"
+LASTEST_VERSION="$(git tag --list | tail -n 1)"
+
+if [ "${VERSION}" = "${LASTEST_VERSION}" ]; then
+    echo "Nothing to release"
+    exit 0
+fi
+
+# Make sure local fork is updated
+git fetch -p --all
+git checkout master
+git merge upstream/master
+
+# Bump version number
+echo ${VERSION} > VERSION
+
+# Generate the package
+make package-clean package
+
+# Sanity check Pulp Smash packages on both Python 2 and Python 3
+for python in python{2,3}; do
+    virtualenv -p ${python} pulpsmash-${python}
+    source pulpsmash-${python}/bin/activate
+    for dist in dist/*.{whl,tar.gz}; do
+        pip install "${dist}"
+        python -m pulp_smash
+        pip uninstall -y pulp_smash
+    done
+    deactivate
+    rm -rf pulpsmash-${python}
+done
+
+# Get the changes from last release and commit
+cat > commit_message <<EOF
+Realease version ${VERSION}
+
+Shortlog of commits since last release:
+
+EOF
+git shortlog "${LASTEST_VERSION}".. | sed "s/^/    /" >> commit_message
+git commit -a -F commit_message
+rm commit_message
+
+# Tag and push to upstream
+git tag ${VERSION}
+git push --tags upstream master
```

Comment:
Can we omit this last step? I do like the idea of being able to look thing over *one* last time before publishing changes. We can put something like this at the end:

```sh
fmt <<EOF
This script has made only local changes: it has updated the VERSION file, generated a new commit, tagged the new commit, and performed a few checks along the way. If you are confident with these changes, you can publish them with commands like the following:
EOF
cat <<EOF

    make publish
    git push --tags origin master && git push --tags upstream master

EOF